### PR TITLE
Support for Language Server Protocol

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do?

[Implemented features](https://docs.google.com/document/d/1zFAz12bgiDfcEF7YDwaj6S2-qWXX1NyHoW-VO7JgiHs/edit?ts=57aae6ab#heading=h.btda58ekv7cj)

Base issue: https://github.com/eclipse/che/issues/1287
### New behavior

Support Language Server protocol

Please review [Che's Contributing Guide](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md) for best practices.

### CODE REVIEW

**Comment by tolusha** [comment]
@evidolob 
@vparfonov

File: wsmaster/che-core-api-agent/src/main/resources/agents/scripts/org.eclipse.che.ls.php.script.sh [hunk 1]
```diff
@@ -0,0 +1,143 @@
+#
```
[the remark this answers is not in the document]
Comment (tolusha) [reply wsmaster/che-core-api-agent/src/main/resources/agents/scripts/org.eclipse.che.ls.php.script.sh:1]: Temporary solution until conception of AgentRegistry is done.

**Comment by vparfonov** [comment]
ok

**Comment by TylerJewell** [comment]
@tolusha @evidolob - this is substantial.  congrats.

**Pull request merged**